lib/trento_web/controllers/v1/settings_controller.ex

```diff
@@ -40,4 +42,56 @@ defmodule TrentoWeb.V1.SettingsController do
 
     json(conn, %{})
   end
+
+  operation :get_api_key_settings,
+    summary: "Get Api key settings",
+    tags: ["Platform"],
+    responses: [
+      ok: {"Api Key settings", "application/json", Schema.Platform.ApiKeySettings},
+      not_found: Schema.NotFound.response()
+    ]
+
+  def get_api_key_settings(conn, _) do
+    with {:ok, api_key_settings} <- Settings.get_api_key_settings() do
+      settings_with_key =
+        Map.put(
+          api_key_settings,
+          :generated_api_key,
+          AuthenticateAPIKeyPlug.generate_api_key!(api_key_settings)
+        )
+
+      render(conn, "api_key_settings.json", %{
+        settings: settings_with_key
+      })
+    end
+  end
+
+  operation :update_api_key_settings,
+    summary: "Updates the Api key settings",
+    tags: ["Platform"],
+    request_body:
+      {"ApiKeySettingsUpdateRequest", "application/json",
+       Schema.Platform.ApiKeySettingsUpdateRequest},
+    responses: [
+      ok: {"Settings saved successfully", "application/json", Schema.Platform.ApiKeySettings},
+      unprocessable_entity: Schema.UnprocessableEntity.response(),
+      not_found: Schema.NotFound.response()
+    ]
+
+  def update_api_key_settings(%{body_params: body_params} = conn, _) do
+    %{expire_at: expire_at} = body_params
+
+    with {:ok, updated_settings} <- Settings.update_api_key_settings(expire_at) do
+      settings_with_key =
+        Map.put(
+          updated_settings,
+          :generated_api_key,
```

Comment:
I would change this `generated_api_key` to `api_key`, but well, up to you. It is not a big deal